packages/shared/src/label-expression-selector/labelExpressionSelector.tsx

```diff
@@ -0,0 +1,424 @@
+  const valueOptions = React.useMemo(
+    () =>
+      (!!key ? labels[key] || [] : []).map((value, i) => (
+        <SelectOption key={i} value={value} />
+      )),
+    [labels, key]
+  );
+
+  // Reset values when label is selected.
+  const onKeyChange = React.useCallback(
+    (selectedKey: string) => {
+      onSelect(index, {
+        key: selectedKey,
+        operator,
+        values: key === selectedKey ? values : [],
+      });
+    },
+    [index, key, operator, values, onSelect]
+  );
+
+  // Reset values when label only operator is selected.
+  const onOperatorChange = React.useCallback(
+    (selectedOperator: Operator) => {
+      onSelect(index, {
+        ...selectedExpression,
+        operator: selectedOperator,
+        values: !isLabelOnlyOperator(selectedOperator) ? values : [],
+      });
+    },
+    [index, selectedExpression, values, onSelect]
+  );
+
+  // Call onSelect to update the state
+  const onValuesChange = React.useCallback(
+    (selectedValues: string[]) => {
+      onSelect(index, {
+        ...selectedExpression,
+        values: selectedValues,
+      });
+    },
+    [index, selectedExpression, onSelect]
+  );
+
+  const isKeyValid = getValidatedProp(isValidationEnabled && !key);
+  const isValuesValid = getValidatedProp(isValidationEnabled && !values.length);
+
+  return (
+    <Grid hasGutter>
+      <GridItem lg={4} sm={4}>
+        <FormGroup
+          label={t('Label')}
+          helperTextInvalid={t('Required')}
+          hasNoPaddingTop
+          isRequired
+          validated={isKeyValid}
+        >
+          <SingleSelectDropdown
+            data-test-id={`key-selection-dropdown-${index}`}
+            id="label-selection-dropdown"
+            placeholderText={t('Select a label')}
+            selectedKey={key}
+            selectOptions={keyOptions}
+            onChange={onKeyChange}
+            hasInlineFilter
+            required
+            validated={isKeyValid}
+            isCreatable
+          />
+        </FormGroup>
+      </GridItem>
+      <GridItem lg={4} sm={4}>
+        <FormGroup
+          label={t('Operator')}
+          helperTextInvalid={t('Required')}
+          isRequired
+          hasNoPaddingTop
+        >
+          <SingleSelectDropdown
+            data-test-id={`operator-selection-dropdown-${index}`}
+            id="operator-selection-dropdown"
+            selectedKey={operator}
+            selectOptions={operatorOptions}
+            onChange={onOperatorChange}
+            hasInlineFilter
+            required
+          />
+        </FormGroup>
+      </GridItem>
+      {!isLabelOnlyOperator(operator) && (
+        <GridItem lg={4} sm={4}>
+          <FormGroup
+            label={t('Values')}
+            helperTextInvalid={t('Required')}
+            hasNoPaddingTop
+            isRequired
+            validated={isValuesValid}
+          >
+            <MultiSelectDropdown
+              data-test-id={`values-selection-dropdown-${index}`}
```

Comment:
`data-test-id` is legacy convention...
```suggestion
              data-test={`values-selection-dropdown-${index}`}
```